### PR TITLE
Add storybook entries for remaining form fields

### PR DESCRIPTION
**Pull request by hschallhorn**
Bumps #15646

### Description
Creates a storybook entry for , `SaveableTextArea`, and `EditableField`. Refactored all from class to functional components.  Added documentation of each.

### Acceptance Criteria
- [ ] Stories exist for the following
    - [ ] [Inline Form](https://www.caseflowdemo.com/styleguide#inline-form)
    - [ ] [Save Long Text](https://www.caseflowdemo.com/styleguide#save-long-text)
    - [ ] [Save Short Text](https://www.caseflowdemo.com/styleguide#save-short-text)
- [ ] Story _AT LEAST_ includes same [documentation](https://storybook.js.org/docs/react/writing-docs/introduction) as the style guide (look for ways to add more!)
- [ ] Story _AT LEAST_ includes same examples ([stories](https://storybook.js.org/docs/react/writing-stories/introduction#how-to-write-stories)) as the style guide (look for ways to add more!)
- [ ] Add proptype documentation via js docstrings to components that are missing them ([these will be automatically pulled by storybook](https://storybook.js.org/docs/react/writing-docs/doc-blocks#argstable))
- [ ] Add [controls](https://storybook.js.org/docs/react/essentials/controls) for args that will change the display of the component

### Testing Plan
1. Run `make storybook` and navigate to http://localhost:6006/?path=/docs/commons-components-searchbar--big-story

### User Facing Changes

 COMPONENT|BEFORE|AFTER|AFTER 2
---| ---|---|---
`InlineForm`|Screen Shot 2020-12-02 at 9 20 54 AM|Screen Shot 2020-12-02 at 9 25 57 AM
`SaveableTextArea`|Screen Shot 2020-12-02 at 9 54 02 AM|Screen Shot 2020-12-02 at 9 53 08 AM|Screen Shot 2020-12-02 at 9 53 15 AM
`EditableField`|Screen Shot 2020-12-02 at 9 54 10 AM|Screen Shot 2020-12-02 at 11 02 40 AM|Screen Shot 2020-12-02 at 11 03 19 AM

### Code Documentation Updates
- [ ] Add or update code comments at the top of the class, module, and/or component.

### Storybook Story
*For Frontend (Presentationa) Components*
* [ ] Add a [Storybook](https://github.com/department-of-veterans-affairs/caseflow/wiki/Documenting-React-Components-with-Storybook) file alongside the component file (e.g. create `MyComponent.stories.js` alongside `MyComponent.jsx`)
* [ ] Give it a title that reflects the component's location within the overall Caseflow hierarchy
* [ ] Write a separate story (within the same file) for each discrete variation of the component

### CODE REVIEW

File: client/app/components/EditableField.jsx [hunk 1]
```diff
@@ -1,110 +1,161 @@
-import React from 'react';
```
Comment (hschallhorn) [inline client/app/components/EditableField.jsx:1]: May be best viewed with "ignore whitespace changes"!

File: client/app/components/EditableField.jsx [hunk 2]
```diff
@@ -1,110 +1,161 @@
...
+    {textDisplay}
+  </div>;
+};
 
 EditableField.defaultProps = {
   type: 'text',
   value: ''
 };
 
 EditableField.propTypes = {
```
Comment (hschallhorn) [inline client/app/components/EditableField.jsx:98]: Alphabetized and documented!

**Review by jcq — APPROVED** [review]
A few comments and/or suggestions, but this looks amazing. I'm so happy to not only have stories for all of these, but for them to also have been rewritten as functional components with all the jsDoc comments? Hooray! 🎉

File: client/app/components/EditableField.jsx [hunk 3]
```diff
@@ -1,110 +1,164 @@
...
+  if (editing) {
+    actionLinks = <span>
+      <Button onClick={onCancelClick} id={`${name}-cancel`} classNames={buttonClasses}>
+        Cancel
+      </Button>&nbsp;|&nbsp;
+      <Button onClick={onSaveClick} id={`${name}-save`} classNames={buttonClasses}>
+        Save
+      </Button>
+    </span>;
+    textDisplay = <input
```
[the remark this answers is not in the document]
Comment (hschallhorn) [reply client/app/components/EditableField.jsx:67]: Should we add [inputProps](https://github.com/department-of-veterans-affairs/caseflow/blob/3784761f15723e3d1a775db3a46be7022494a8dc/client%2Fapp%2Fcomponents%2FTextField.jsx#L96) as well or was that a text field specific addition?

[the remark this answers is not in the document]
Comment (jcq) [reply client/app/components/EditableField.jsx:65]: Oh yeah, that should be a standard thing to pass through as well. Good catch!

**Pull request merged**